hugegraph-api/src/main/java/com/baidu/hugegraph/api/job/GremlinAPI.java

```diff
@@ -148,7 +152,21 @@ public String gremlin() {
 
         public String name() {
             // Get the first line of script as the name
-            return this.gremlin.split("\r\n|\r|\n", 2)[0];
+            String firstLine = this.gremlin.split("\r\n|\r|\n", 2)[0];
+            try {
+                byte[] srcBytes = firstLine.getBytes(CHARSET);
+                if (srcBytes.length > MAX_NAME_LENGTH) {
+                    byte[] dstBytes = new byte[MAX_NAME_LENGTH];
+                    System.arraycopy(srcBytes, 0, dstBytes, 0, MAX_NAME_LENGTH);
```

Comment:
may cause encoding exception if the end byte is truncated
see: [decoder.onMalformedInput(CodingErrorAction.IGNORE);](https://stackoverflow.com/questions/3576754/truncating-strings-by-bytes)